server/portal/apps/onboarding/steps/project_membership.py

```diff
@@ -1,77 +1,209 @@
+                        username=self.user.username,
+                        reason=reason
+                    )
+                )
+                raise e
+
+    def close_project_request(self):
+        ticket_id = None
+        for event in self.events:
+            if event.data and "ticket" in event.data:
+                ticket_id = event.data["ticket"]
+        tracker = self.get_tracker()
+        request_text = """Your request for membership on the {project} project has been
+        granted. Please login at {base_url}/onboarding/setup to continue setting up your account.
+        """.format(
+            project=self.project['title'],
+            base_url=settings.WH_BASE_URL
+        )
+        if tracker.login():
+            tracker.reply(ticket_id, text=request_text)
+            tracker.comment(
+                ticket_id,
+                text="User has been added to the {project} TAS Project (GID {gid}) via {base_url}".format(
+                    project=self.project['title'],
+                    gid=self.project['gid'],
+                    base_url=settings.WH_BASE_URL
+                )
+            )
+            # This call fails with no explanation (BadRequest, but no reason)
+            tracker.edit_ticket(ticket_id, Status='resolved')
+        else:
+            self.fail(
+                "The portal was unable to close RT Ticket {ticket_id}".format(
+                    ticket_id=ticket_id
+                )
+            )
 
-        has_required_project = any(p['chargeCode'] in required_projects for p in projects)
+    def process(self):
+        if self.is_project_member():
+            self.complete("You have the required project membership to access this portal.")
+        else:
+            self.state = SetupState.USERWAIT
+            self.log(
+                "Please confirm your request to use this portal.",
+            )
+
+    def client_action(self, action, data, request):
+        if action == "user_confirm":
+            self.send_project_request(request)
+            return
 
-        if has_required_project:
-            logger.info("User is on a required project")
-            self.complete("You are on a required project needed to access this portal")
+        if request.user.is_staff and action == "staff_approve":
+            try:
+                self.add_to_project()
+                self.close_project_request()
+                self.complete(
+                    "Portal access request approved by {user}".format(
+                        user=request.user.username
+                    )
+                )
+            except Exception as err:
+                logger.exception(msg="Error during staff_approve on ProjectMembershipStep")
+                logger.error(err.args)
+                self.fail(
+                    "An error occurred while trying to add this user to the project"
+                )
+        elif action == "staff_deny":
+            self.fail(
```

Comment:
I think we should reply to the ticket and close it here like how you do during ` action == "staff_approve" ` case.  maybe refactor `close_project_request` so it can have a boolean parameter